shutdown debugging

mmaslankaprv opened the pull request:
TBD
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->
- none

mmaslankaprv commented:
/ci-repeat 1
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
skip-redpanda-build
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
skip-redpanda-build
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
skip-redpanda-build
tests/rptest/tests/random_node_operations_test.py

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--8c18cd5a-33e0-408c-8546-4b0356a0dc4d-->
<!--TEST RESULTS-->
<!--ae97d78a-845c-49c4-a0e3-9ede58ed090e-->
<!--TEST RESULTS-->
<!--d5733c4b-3dae-49e8-b581-5f1b0075a08d-->
<!--TEST RESULTS-->
<!--a51b9e1e-d77a-4f1a-ac00-921445ed3957-->
<!--TEST RESULTS-->
<!--f632999f-6489-4f64-a637-d195b5d7950e-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/69952">build#69952</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                                  | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:-------------------------|:---------------------|:------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 1, "compaction_mode": "chunked_sliding_window", "enable_failures": true, "mixed_versions": false, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/69952#01985bba-cdca-4a60-aeb9-20c97aa45815 | FLAKY         | 19/21    | upstream reliability is '99.20634920634922'. current run reliability is '90.47619047619048'. drift is 8.73016 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/70004">build#70004</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                                  | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:-------------------------|:---------------------|:------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 2, "compaction_mode": "adjacent_merge", "enable_failures": false, "mixed_versions": true, "with_iceberg": false}         | integration | https://buildkite.com/redpanda/redpanda/builds/70004#01985fa5-7c99-4dad-ba1d-ca5646caf5f2 | FLAKY         | 20/21    | upstream reliability is '99.23954372623575'. current run reliability is '95.23809523809523'. drift is 4.00145 and the allowed drift is set to 50. The test should PASS |
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 1, "compaction_mode": "chunked_sliding_window", "enable_failures": false, "mixed_versions": true, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/70004#01985fa5-7ca7-489e-bf68-7cc836d5b813 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS              |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/70012">build#70012</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                           | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  |
|:-------------------------|:---------------------|:-----------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 1, "compaction_mode": "sliding_window", "enable_failures": false, "mixed_versions": false, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/70012#019860aa-d7db-4834-95b7-9e0dd7c5aa70 | FLAKY         | 18/21    | upstream reliability is '99.01639344262296'. current run reliability is '85.71428571428571'. drift is 13.30211 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/70069">build#70069</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:-------------------------|:---------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 1, "compaction_mode": "chunked_sliding_window", "enable_failures": true, "mixed_versions": false, "with_iceberg": true} | integration | https://buildkite.com/redpanda/redpanda/builds/70069#019865ab-5648-4c7e-927c-102efeae8794 | FLAKY         | 19/21    | upstream reliability is '99.28825622775801'. current run reliability is '90.47619047619048'. drift is 8.81207 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/70211">build#70211</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                          | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    |
|:-------------------------|:---------------------|:----------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 2, "compaction_mode": "sliding_window", "enable_failures": true, "mixed_versions": false, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/70211#01987961-5ec8-440c-a025-4209ba1b85b4 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
skip-redpanda-build
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
skip-redpanda-build
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv committed:
wip

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 6
skip-redpanda-build
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv committed:
c/leaders_table: wire in controller abort source

Controller input is stopped in the early stage of Redpanda shutdown
process. This way all the request are failed early when Redpanda is
shutting down. Passing in a `controller` abort source to the
`partition_leaders_table` will allow us to interrupt all fibers waiting
for the leaders instead of having to wait for the timeout to expire.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
tests/rptest/tests/random_node_operations_test.py

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
skip-redpanda-build
tests/rptest/tests/random_node_operations_test.py
